src/plugins/es_ui_shared/docs/docs/form_lib/core/use_form_hook.md

```diff
@@ -0,0 +1,260 @@
+---
+id: use_form_hook
+title: useForm()
+sidebar_label: useForm()
+---
+
+**Returns:** [`FormHook`](form_hook.md)
+
+Use the `useForm` hook to declare a new form object. As we have seen in the ["Getting started"](about.md), you can use it without any additional configuration. It does accept an optional `config` object with the following configuration (all parameters are optionals).
+
+## Configuration
+
+### onSubmit(data, isValid)
+
+**Arguments:** `data: T, isValid: boolean`  
+**Returns:** `Promise<void>`
+
+The `onSubmit` handler is executed when calling `form.submit()`. It receives the form data and a boolean for the validity of the form.
+Meanwhile the form is being submitted, the form `isSubmitting` state will be set to `true`. This can be useful to change the state of the submit button in the UI.
```

Comment:
```suggestion
When the form is submitted `isSubmitting` will be set to `true` then back to `false` after the `onSubmit` handler has run. This can be useful to change the state of the submit button in the UI.
```